statsd/receiver.go

```diff
@@ -145,21 +143,34 @@ func (mr *metricReceiver) handleMessage(ctx context.Context, addr net.Addr, msg
 				triedToGetTags = true
 				additionalTags = mr.getAdditionalTags(addr.String())
 			}
-			if len(additionalTags) > 0 {
+			if metric != nil {
+				numMetrics++
+				metric.Tags = append(metric.Tags, mr.tags...)
 				metric.Tags = append(metric.Tags, additionalTags...)
+				err = mr.handler.DispatchMetric(ctx, metric)
+			} else {
```

Comment:
Ok, in that case I'll add an `else` statement as well with a panic to catch bugs. Good idea.